### ISSUE TITLE
Fluid Files Not Found Properly

### ISSUE THREAD

**Issue by dnwillia-ansys** [usnistgov/REFPROP-issues#18]
In version 9.4.4.44 (001) one of my tests is failing where I specify the fluids file by a relative path like this:

`call setpath('../../../refprop')`

and then call setup with a single fluid like this:

`call setup(1, 'fluids/water', 'hmx.bnc', 'DEF', icode, msg)`

The setup fails.  This was working in previous versions I've tried.

**Comment by ianhbell** [comment]
Interesting - I'll add a test for this.  I never use relative paths, I always use language facilities to convert relative paths to absolute paths (at runtime), as the relative paths are much more fragile.

Also, you should definitely not be prepending ``fluids/`` to the fluid name, maybe that is actually the problem here.

**Comment by dnwillia-ansys** [comment]
I don't know of any language facilities for converting relative to absolute paths in C, C++ or Fortran.  Let me know if I'm missing something.  I use relative paths because I unit test the API and the fluid files are sitting in that location relative to the unit test exe.

Regarding prepending `fluids` to the path I only do that because the documentation tells me I can do that.  From manual.txt:

```
      subroutine SETUP (ncomp,hfiles,hfmix,hrf,ierr,herr)
c
c  define models and initialize arrays
c
c  A call to this routine is required.
c
c  inputs:
c    ncomp--number of components (1 for pure fluid) [integer]
c           if called with ncomp=-1, the version number*100 will be returned in ierr
c   hfiles--array of file names specifying fluid/mixture components
c           [character*255 variable] for each of the ncomp components;
c           e.g., :fluids:r134a.fld (Mac) or fluids\r134a.fld (DOS) or
c           [full_path]/fluids/r134a.fld (UNIX) 
```
So perhaps you need to update the documentation?

**Comment by ianhbell** [comment]
In C++, with boost, you can do: https://github.com/ianhbell/REFPROP-tests/blob/master/include/REFPROPtests/utils.h#L113-L116 .  It’s true that C and FORTRAN probably don’t have native functionality for that.  But most high-level languages do.

I always lay out my folders with the dll and then the ``FLUIDS`` and ``MIXTURES`` folders as subfolders relative to the DLL, mirroring the layout on windows.  

I'll investigate the relative paths issue further with Eric.  Thanks for the report.

**Comment by dnwillia-ansys** [comment]
Thanks Ian.   I see what you mean now.  I didn't really consider boost a language facility because not all of it has been incorporated into the standard library yet, [but they are working on adding filesystem](http://en.cppreference.com/w/cpp/filesystem).

**Comment by ianhbell** [comment]
Yes, boost is not technically part of the language, and for a very(!) long time I resisted using it, but nowadays with Cmake + boost (https://github.com/Orphis/boost-cmake) it's seamless, and my reservations are gone.  That's what I use in the tests repository.

**Comment by dnwillia-ansys** [comment]
I tried the latest version, 9.4.4.44 (003), and it still has the same issue.

**Comment by ianhbell** [comment]
I think you might need the even-newer 10.b installer, though not sure if this bug will be resolved
